### PR TITLE
Update proxy pipeline to staging and production instanced pipelines

### PR DESCRIPTION
**Pull request by apburnes**
## Changes proposed in this pull request:
- Refactors ci pipeline to use instanced pipelines
- Add self setting pipeline task to set pipeline on update to staging or main branch
- Add deploy docs in README

## security considerations
None

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -220,114 +226,28 @@ jobs:
...
 
-  - name: src-staging
-    type: git
-    icon: github
-    source:
-      uri: ((git-base-url))/((proxy-repository-path))
-      branch: staging
-
-  - name: src-production
+  - name: src-((deploy-env))
```
Comment (hursey013) [inline ci/pipeline.yml:237]: It didn't complain about this? I swore this was giving me trouble before

File: README.md [hunk 2]
```diff
@@ -61,6 +61,61 @@ variable will need to be updated in the `manifest.yml`.
   docker-compose run --rm app npm run test:integration
 ```
 
+## Continuous Integration
+The proxy uses Councourse CI to run tests and deploy to different environments in the cloud.gov Pages organization. The pipeline is defined in the [`ci/pipeline.yml`]('./ci/pipeline.yml') file and supporting CI scripts are found in the [`ci`]('./ci') directory. This pipeline is using Concourse's [`instanced pipeline`](https://concourse-ci.org/instanced-pipelines.html) feature to minimize boilerplate configuration when declaring tasks and resources for each deployment environment.
```
Comment (hursey013) [inline README.md:65]:
```suggestion
The proxy uses Concourse CI to run tests and deploy to different environments in the cloud.gov Pages organization. The pipeline is defined in the [`ci/pipeline.yml`]('./ci/pipeline.yml') file and supporting CI scripts are found in the [`ci`]('./ci') directory. This pipeline is using Concourse's [`instanced pipeline`](https://concourse-ci.org/instanced-pipelines.html) feature to minimize boilerplate configuration when declaring tasks and resources for each deployment environment.
```

File: README.md [hunk 3]
```diff
@@ -61,6 +61,61 @@ variable will need to be updated in the `manifest.yml`.
...
+
+### Pipeline instance variables
+Two instances of the pipeline are set for the `staging` and `production` environments. Instance variables are used to fill in Concourse pipeline parameter variables baring the same name as the instance variable. See more on [Concourse vars](https://concourse-ci.org/vars.html).  Each instance of the pipeline has two instance variables associated to it: `deploy-env` & `git-branch`.
```
Comment (hursey013) [inline README.md:68]:
```suggestion
Two instances of the pipeline are set for the `staging` and `production` environments. Instance variables are used to fill in Concourse pipeline parameter variables barring the same name as the instance variable. See more on [Concourse vars](https://concourse-ci.org/vars.html).  Each instance of the pipeline has two instance variables associated to it: `deploy-env` & `git-branch`.
```

File: README.md [hunk 4]
```diff
@@ -61,6 +61,61 @@ variable will need to be updated in the `manifest.yml`.
...
+|**`((slack-channel))`**| Slack channel | :x:|
+|**`((slack-username))`**| Slack username | :x:|
+|**`((slack-icon-url))`**| Slack icon url | :x:|
+|**`((slack-webhook-url))`**| Slack webhook url | :x:|
+|**`((git-base-url))`**|The base url to the git server's HTTP endpoint|:x:|
+|**`((proxy-repository-path))`**|The url path to the repository|:x:|
+|**`((gh-access-token))`**| The Github access token|:x:|
+
+### Setting up the pipeline
+The pipeline and each of it's instances will only needed to be set once per instance to create the initial pipeline. After the pipelines are set, updates to the respective `git-branch` source will automatically set the pipeline with any updates. See the [`set_pipeline` step](https://concourse-ci.org/set-pipeline-step.html) for more information. Run the following command with the fly CLI to set a pipeline instance:
```
Comment (hursey013) [inline README.md:97]:
```suggestion
The pipeline and each of it's instances will only need to be set once per instance to create the initial pipeline. After the pipelines are set, updates to the respective `git-branch` source will automatically set the pipeline with any updates. See the [`set_pipeline` step](https://concourse-ci.org/set-pipeline-step.html) for more information. Run the following command with the fly CLI to set a pipeline instance:
```

**Review by hursey013 — APPROVED** [review]
This is awesome, love the documentation as well! 🎉

File: README.md (hunk 3 above)
Comment (apburnes) [reply README.md:68]: I think I meant `bearing`

Comment (apburnes) [reply README.md:68]: I think I meant `bearing`

Comment (apburnes) [reply README.md:68]: I think meant `bearing`

File: ci/pipeline.yml (hunk 1 above)
Comment (apburnes) [reply ci/pipeline.yml:237]: Maybe due to the `instance-var` feature being enabled recently.

File: README.md (hunk 3 above)
Comment (apburnes) [reply README.md]: Hahah. I don't know what happened ☝️ but I did mean `bearing`.

**Pull request merged**